### PR TITLE
Refactor facet filter panel so that its easy to add extra drilldown filt...

### PR DESCRIPTION
**Pull request by jonescc**
...ers

Note, this is a reorganisation of code only, no new functionality has been added.

Adding extra drilldown filters should be simple after this change.

### CODE REVIEW

**Comment by danfruehauf** [comment]
I'm having a look.

File: src/test/javascript/portal/search/FacetFilterPanelSpec.js [hunk 1]
```diff
@@ -7,12 +7,12 @@
 describe("Portal.search.FacetFilterPanel", function() {
 
     var searcher;
-    var selectionPanel;
-
+    var filterPanel;
+
```
Comment (danfruehauf) [inline src/test/javascript/portal/search/FacetFilterPanelSpec.js]: Spaces here.

File: src/test/javascript/portal/search/FacetFilterPanelSpec.js [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
     var searcher;
-    var selectionPanel;
-
+    var filterPanel;
+    
     beforeEach(function() {
         searcher = new Portal.service.CatalogSearcher();
-
-        selectionPanel = new Portal.search.FacetFilterPanel({
+
```
Comment (danfruehauf) [inline src/test/javascript/portal/search/FacetFilterPanelSpec.js]: Spaces also here.

File: web-app/js/portal/search/FacetDrilldownPanel.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ *
+ */
+
+Ext.namespace('Portal.search');
+
+Portal.search.FacetDrilldownPanel = Ext.extend(Ext.tree.TreePanel, {
+    constructor: function(cfg) {
+    	cfg = cfg || {};
+    	
+	    Ext.apply(cfg, {
```
Comment (danfruehauf) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: Bad indent for this whole block.

File: web-app/js/portal/search/FacetDrilldownPanel.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName));
+    	} else {
+    		this.setRootNode(new Ext.tree.TreeNode());
+    	}
+    },
+
+    _onCheckChange: function(node) {
+        this._selectedNodeValueHierarchy = node.parentNode.toValueHierarchy();
+    	this.fireEvent('drilldownchange');
```
Comment (danfruehauf) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: Indent with spaces here, and to 4 spaces.

File: web-app/js/portal/search/FacetDrilldownPanel.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	
+    getDrilldownPath: function() {
+    	return this._selectedNodeValueHierarchy;
+    },
+
+    _initRootNode: function() {
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName));
+    	} else {
+    		this.setRootNode(new Ext.tree.TreeNode());
```
Comment (danfruehauf) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: Also here, fix the indent.

File: web-app/js/portal/search/FacetDrilldownPanel.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        }, this);
+    },
+	
+    getDrilldownPath: function() {
+    	return this._selectedNodeValueHierarchy;
+    },
+
+    _initRootNode: function() {
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName));
```
Comment (danfruehauf) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: And here.

File: web-app/js/portal/search/FacetDrilldownPanel.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+    _onSelectionChange: function(selectionModel, node) {
+        this._selectedNodeValueHierarchy = node.toValueHierarchy();
+        trackFacetUsage(this.facetName, node.attributes.value);
+        this.fireEvent('drilldownchange');
+    },
+
+    _onSearchComplete: function() {
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName))
```
Comment (danfruehauf) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: Also indented with tabs for some reason.

Comment (jonescc) [inline web-app/js/portal/search/FacetDrilldownPanel.js]: Looks like my eclipse ide has the wrong setting for indenting for javascript files - I'll fix it and the problems its caused.

**Comment by danfruehauf** [comment]
@jonescc Looking fine. I don't see anything problematic here. It's a good and necessary refactor. Just fix the indents and squash them with the previous commit and it should be good to go.

**Comment by jonescc** [comment]
@danfruehauf whitespace issues fixed

**Comment by danfruehauf** [comment]
Not entirely. Still tabs in many places (`git show 6b341242cc63 | grep '<TAB>'`).

I'll fix it.

**Comment by jonescc** [comment]
? git show 6b341242cc63 | grep '<TAB>'   returns nothing for me

**Comment by danfruehauf** [comment]
You need to quote a tab character. It's done by doing `Ctrl+V` then hitting `<TAB>`.

**Comment by danfruehauf** [comment]
Don't worry though, I fixed those for you.

**Comment by jonescc** [comment]
OK, I see them now.  Shouldn't be an issue next time - fixed my IDE settings and now displaying whitespace characters so I can see them.

**Pull request merged**